features/step_definitions/restrict-pr.js

```diff
@@ -20,6 +20,11 @@ Before(
         this.jobName = 'main';
         this.pullRequestNumber = null;
         this.pipelines = {};
+
+        return Promise.all([
+            github.removeBranch(this.repoOrg, this.repoName, 'test-branch-PR'),
+            github.removeBranch('screwdriver-cd', this.repoName, 'test-branch-PR')
```

Comment:
should this line be deleted?

Reply:
This line should not be deleted.
This line has been moved from the After hook to the Before hook.
It solves the problem of PR remaining by deleting the previous PR first.

Reply:
In these scenarios, two repositories, screwdriver-cd-test and screwdriver-cd, are referenced so that both are needed.

https://github.com/screwdriver-cd/screwdriver/blob/dc27788d7e1ad052c0b26efbd0883f669363d7c2/features/restrict-pr.feature#L13-L59

Reply:
Although we need to delete a branch from another org, we should avoid putting it literally to step_definitions.

Reply:
How about moving the source branch removing to before this?
https://github.com/screwdriver-cd/screwdriver/blob/53e9dbcf2970d736476c7491d8483ce9640c336f/features/step_definitions/restrict-pr.js#L62-L64

Reply:
If the problem can't solve by removing it after the exact value, we should consider adding a step to remove it into the definition/feature.

Reply:
I have adopted the following comment and modified it.
https://github.com/screwdriver-cd/screwdriver/pull/2899#discussion_r1268937985

Reply:
my :+1: for new changes